kclvm/tools/src/LSP/src/main_loop.rs

```diff
@@ -1,8 +1,86 @@
 use crate::config::Config;
 use crate::state::LanguageServerState;
+use crate::{capabilities, util};
+use clap::Command;
 use lsp_server::Connection;
 
+#[allow(dead_code)]
 /// Runs the main loop of the language server. This will receive requests and handle them.
 pub(crate) fn main_loop(connection: Connection, config: Config) -> anyhow::Result<()> {
     LanguageServerState::new(connection.sender, config).run(connection.receiver)
 }
+
+#[allow(dead_code)]
+/// Main entry point for the language server
+fn run_server() -> anyhow::Result<()> {
+    // Setup IO connections
+    let (connection, io_threads) = lsp_server::Connection::stdio();
+    // Wait for a client to connect
+    let (initialize_id, initialize_params) = connection.initialize_start()?;
+
+    let initialize_params =
+        util::from_json::<lsp_types::InitializeParams>("InitializeParams", initialize_params)?;
+
+    let server_capabilities = capabilities::server_capabilities(&initialize_params.capabilities);
+
+    let initialize_result = lsp_types::InitializeResult {
+        capabilities: server_capabilities,
+        server_info: Some(lsp_types::ServerInfo {
+            name: String::from("kcl-language-server"),
+            version: None,
+        }),
+    };
+
+    let initialize_result = serde_json::to_value(initialize_result)
+        .map_err(|_| anyhow::anyhow!("Initialize result error"))?;
+
+    connection.initialize_finish(initialize_id, initialize_result)?;
+
+    let config = Config::default();
+    main_loop(connection, config)?;
+    io_threads.join()?;
+    Ok(())
+}
+
+#[allow(dead_code)]
+#[derive(Copy, Debug, Clone, PartialEq, Eq)]
+enum ExitStatus {
+    Success,
+    Error,
+}
+
+pub(crate) fn _main(args: Vec<String>) -> Result<(), anyhow::Error> {
```

Comment:
Use the type annotation for the argument `args: I where I: IntoIterator<Item = T>`.

Besides, why use the `pub(crate)` keyword for a function name `_main`. `_main` seems like a private function.